components/calendar/src/cal/chinese.rs

```diff
@@ -575,47 +574,35 @@ impl<R: Rules> DateFieldsResolver for LunarChinese<R> {
         month_code: types::ValidMonthCode,
         options: DateFromFieldsOptions,
     ) -> Result<u8, MonthCodeError> {
-        match year.parse_month_code(month_code) {
-            ComputedOrdinalMonth::Exact(val) => Ok(val),
-            ComputedOrdinalMonth::LeapToNormal(val) => {
-                if options.overflow == Some(Overflow::Constrain) {
-                    Ok(val)
-                } else {
-                    Err(MonthCodeError::NotInYear)
-                }
-            }
-            ComputedOrdinalMonth::NotFound => Err(MonthCodeError::NotInCalendar),
+        // 14 is a sentinel value, greater than all other months, for the purpose of computation only;
```

Comment:
issue: I don't find this to be more readable than what we had before. `parse_month_code` wasn't super readable in the first place, but at least this method was very clear in its behavior when it called it. this merges a readable method with an unreadable one and produces a larger, more complex unreadable one IMO.

possible to resolve by documenting the code more.